### PR TITLE
Changes to MasterWindow, tests etc

### PR DESCRIPTION
**Pull request by siemiatj**
Fixes error in #2517, where only the latest change was stored in redux. #2604 fixes regression caused by changes to text inputs, rewrites some of our actions to be fully async and adds some tests.

Props used for `TableItem` are just JSON.stringified props of a real component before/after changes.

### CODE REVIEW

**Comment by siemiatj** [comment]
https://recordit.co/DzjhTQSdru

File: src/utils/websockets.js [hunk 1]
```diff
@@ -34,10 +38,33 @@ export function connectWS(topic, onMessageCallback) {
...
+      brokerURL: config.WS_URL,
+      // debug: function(str) {
+      //   console.log('debug: ', str);
+      // },
+      reconnectDelay: 5000,
+      heartbeatIncoming: 4000,
+      heartbeatOutgoing: 4000,
+    });
+
+    if (process.env.JEST_WORKER_ID === undefined) {
```
Comment (petrican) [inline src/utils/websockets.js]: quotes around undefined

File: src/utils/websockets.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import SockJs from 'sockjs-client';
-import Stomp from 'stompjs/lib/stomp.min.js';
+import SockJS from 'sockjs-client';
+import { Client } from '@stomp/stompjs';
+
+function socketFactory() {
+  return new SockJS(config.WS_URL);
```
Comment (petrican) [inline src/utils/websockets.js]: Where is the config defined ? Here this is undefined

File: src/utils/websockets.js [hunk 3]
```diff
@@ -34,10 +38,33 @@ export function connectWS(topic, onMessageCallback) {
   };
 
   const connect = () => {
-    this.sock = new SockJs(config.WS_URL);
-    this.sockClient = Stomp.Stomp.over(this.sock);
-    this.sockClient.debug = null;
-    this.sockClient.connect({}, subscribe);
+    this.sockClient = new Client({
+      brokerURL: config.WS_URL,
```
Comment (petrican) [inline src/utils/websockets.js]: same as above comment

Comment (siemiatj) [reply src/utils/websockets.js]: config is a global in `config.js`

File: src/utils/websockets.js (hunk 2 above)
Comment (siemiatj) [reply src/utils/websockets.js]: config is a global in `config.js`

File: src/utils/websockets.js (hunk 1 above)
Comment (siemiatj) [reply src/utils/websockets.js]: it's not `typeof` so it won't be a string

**Pull request merged**